File: src/main/java/org/mtransit/android/data/NewsProviderProperties.java
package org.mtransit.android.data;

public class NewsProviderProperties {

	private String targetAuthority;
	private String authority;

	public NewsProviderProperties(String authority, String targetAuthority) {
		this.authority = authority;
		this.targetAuthority = targetAuthority;
	}

	public String getTargetAuthority() {
		return this.targetAuthority;
	}

	public String getAuthority() {
		return authority;
	}

	@Override
	public String toString() {
		return new StringBuilder().append(NewsProviderProperties.class.getSimpleName()).append('{') //
				.append("authority:").append(authority).append(',') //
				.append("targetAuthority:").append(targetAuthority).append(',') //
				.append('}').toString();
	}
}

File: src/main/java/org/mtransit/android/data/NewsProviderProperties.kt
package org.mtransit.android.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(
    tableName = "news_provider_properties",
)
data class NewsProviderProperties(
    @PrimaryKey
    @ColumnInfo(name = "authority")
    override val authority: String,
    @ColumnInfo(name = "target_authority")
    override val targetAuthority: String,
    @ColumnInfo(name = "pkg")
    override val pkg: String,
) : ITargetedProviderProperties
